internal/daemon/cache/repository_refresh_test.go

```diff
@@ -29,6 +25,14 @@ func noopRetrievalFn[T any](context.Context, string, string) ([]T, error) {
 	return nil, nil
 }
 
+// staticRetrievalFn returns a function that satisfies the With*RetrievalFn
```

Comment:
It's not clear what the With*RetrievalFn might be since it's not searchable term
in the code base.  Perhaps: returns a function that satisifies a "retrieval
function" like TargetRetrievalFunc or SessionRetrievalFunc.